### DISCUSSION TITLE
[FEATURE]: Enhancement of release

### DISCUSSION THREAD

**Discussion by Aju100** [envless/envless#323] [Ideas]
### Description

Hello @dahal , Saw out the manual process of [release notes](https://github.com/envless/envless/releases/tag/v0.1.1) at Envless. It can be automated with reference to conventional commits with semantic version. Release notes can be automated and make a PR for it. Few things regarding new contributors, we can update on the same PR manually.

Ref:
https://www.conventionalcommits.org/en/v1.0.0/
https://github.com/Aju100/ultimate-actix-web/pull/2

### Possible solution

Release please might be good solution out here.
Source: https://github.com/googleapis/release-please

Implementation and poc:
https://github.com/Aju100/ultimate-actix-web/releases/tag/v0.2.0

**Comment by dahal** [comment]
Ah, this is interesting. Lets brainstorm some workflow. The current flow we have implemented:-

* Def follow the conventional commit patterns.
* Merge to `main` does not trigger build/deploy to prod.
* Build/deploy is only triggered when releases are created and published.
* During build, it also generates static changelog page as you can see here
https://envless.dev/changelog

**Comment by dahal** [comment]
I like your proposal @Aju100 , please let us know if you want to work on this and we will assign it to you.

**Comment by Aju100** [comment]
@dahal Dai do we need to have pre-release options or not. Also I wanted to know when can we create the release notes ?

**Comment by dahal** [comment]
> @dahal Dai do we need to have pre-release options or not. Also I wanted to know when can we create the release notes ?

@Aju100 does #315 satisfy the pre-release option. If not we def should. I am currently going through with some merges to see what happens and how it works. As far as release notes (changelogs) should/are also created when creating the release tags.

**Comment by dahal** [comment]
So, looks like it creates a regular release like #317 , what would the pre-release workflow like? How do we know which one is going to be release and which one a pre-release?

**Comment by Aju100** [comment]
@dahal , if we set up the pre-release workflow here with release please github actions , it will set the release for pre-major or pre-release version which eventually marked as pre-release.

If we wanted to have pre-release for specific branch lets say as "dev" and and release on prod/main/master. Google github actions haven't done yet so far for this feature.
Please have a look of this thread.
- https://github.com/google-github-actions/release-please-action/issues/687

Let me know about this should we enable pre-release option or not ?  cc @dahal

**Comment by dahal** [comment]
~Sounds good. I think this flow makes sense. Let's have a `canary` branch for pre-releases and `main`~.

Now that I think about it, it's going add additional complicity if we want to maintain `canary` branch. How about this flow:-

* Lets not create pre-release
* Lets just have regular `unpublished` release.
* When this unpublished release's status changes to published, it should trigger a deploy (to vercel for now). For this we need to modify this PR to deploy work with turbo and deploy both platform and docs
https://github.com/envless/envless/pull/237

@Aju100

**Comment by Aju100** [comment]
Got it @dahal what about the containerization one and deployment for the container's image one? I am sure #291 or similar future PR will eventually get merged later on.

**Comment by dahal** [comment]
Yes, we have to add more actions later on when the referenced PR is merged.
